File: 2016/day2/pa.lua
local current = { x = 2, y = 2 }
local pad = {
	{ 1, 2, 3 },
	{ 4, 5, 6 },
	{ 7, 8, 9 },
}

local function clamp(val, min, max)
	if val < min then
		return min
	end
	if val > max then
		return max
	end
	return val
end

for line in io.lines("./input.txt") do
	for i = 1, #line do
		local char = line:sub(i, i)

		if char == "U" then
			current.x = clamp(current.x - 1, 1, #pad)
		elseif char == "D" then
			current.x = clamp(current.x + 1, 1, #pad)
		elseif char == "L" then
			current.y = clamp(current.y - 1, 1, #pad[1])
		elseif char == "R" then
			current.y = clamp(current.y + 1, 1, #pad[1])
		end
	end

	print(pad[current.x][current.y])
end

File: 2016-py/day2/pa.py
def clamp(value, minimum, maximum):
    return max(minimum, min(value, maximum))


def main():
    current = [1, 1]  # starts on 5

    pad = [
        [1, 2, 3],
        [4, 5, 6],
        [7, 8, 9],
    ]

    with open("input.txt", "r") as fin:
        for line in fin:
            line = line.strip()

            for char in line:
                if char == "U":
                    current[0] = clamp(current[0] - 1, 0, len(pad) - 1)
                elif char == "D":
                    current[0] = clamp(current[0] + 1, 0, len(pad) - 1)
                elif char == "L":
                    current[1] = clamp(current[1] - 1, 0, len(pad[0]) - 1)
                elif char == "R":
                    current[1] = clamp(current[1] + 1, 0, len(pad[0]) - 1)

            print(pad[current[0]][current[1]])


if __name__ == "__main__":
    main()

